classes/Builder/Payload/OrderPayloadBuilder.php

```diff
@@ -0,0 +1,299 @@
+            $this->getPayload()->setItems($node);
+        }
+    }
+
+    /**
+     * Build the paypal items node
+     *
+     * @return array
+     */
+    public function buildItemsNode()
+    {
+        $node = [];
+
+        $totalProductsWithoutTax = 0;
+        $totalTax = 0;
+
+        foreach ($this->cart['products'] as $product => $value) {
+            $paypalItem = [];
+
+            $paypalItem['name'] = $this->truncate($value['name'], 127);
+            $paypalItem['description'] = $this->truncate(strip_tags($value['description_short']), 127);
+            $paypalItem['sku'] = $this->truncate($value['unity'], 127);
+            $paypalItem['unit_amount']['currency_code'] = $this->cart['currency']['iso_code'];
+            $paypalItem['unit_amount']['value'] = $value['total'] / $value['quantity'];
+            $paypalItem['tax']['currency_code'] = $this->cart['currency']['iso_code'];
+            $paypalItem['tax']['value'] = ($value['total_wt'] - $value['total']) / $value['quantity'];
+            $paypalItem['quantity'] = $value['quantity'];
+            $paypalItem['category'] = $value['is_virtual'] === '1' ? 'DIGITAL_GOODS' : 'PHYSICAL_GOODS';
+
+            $totalProductsWithoutTax += $paypalItem['unit_amount']['value'] * $value['quantity'];
+            $totalTax += $paypalItem['tax']['value'] * $value['quantity'];
+
+            $node['items'][] = $paypalItem;
+        }
+
+        $this->getPayload()->setItems($node);
+
+        return [
+            'totalProductsWithTax' => $totalProductsWithoutTax,
+            'totalTax' => $totalTax,
+        ];
+    }
+
+    /**
+     * Build the amount breakdown node
+     */
+    public function buildAmountBreakdownNode()
+    {
+        $totals = $this->buildItemsNode();
+
+        $node['amount']['breakdown'] = [
+            'item_total' => [
+                'currency_code' => $this->cart['currency']['iso_code'],
+                'value' => $totals['totalProductsWithTax'],
+            ],
+            'shipping' => [
+                'currency_code' => $this->cart['currency']['iso_code'],
+                'value' => $this->cart['cart']['shipping_cost'],
+            ],
+            'tax_total' => [
+                'currency_code' => $this->cart['currency']['iso_code'],
+                'value' => $totals['totalTax'],
+            ],
+        ];
+
+        $this->getPayload()->setItems($node);
+
+        // define by default the handling at 0
+        $handlingValue = 0;
+
+        // set handling cost id needed -> principally used in case of gift_wrapping
+        if (isset($this->cart['cart']['subtotals']['gift_wrapping'])) {
+            $handlingValue += $this->cart['cart']['subtotals']['gift_wrapping']['amount'];
+
+            $node['amount']['breakdown']['handling'] = [
+                'currency_code' => $this->cart['currency']['iso_code'],
+                'value' => $handlingValue,
+            ];
+
+            $this->getPayload()->setItems($node);
```

Comment:
Unneeded, you already have a setItems at the end of this method

Reply:
Removed.